nativelib/src/main/resources/scala-native/gc/immix/ThreadManager.h

```diff
@@ -0,0 +1,21 @@
+#ifndef IMMIX_THREADMANAGER_H
+#define IMMIX_THREADMANAGER_H
+
+#include <pthread.h>
+#include "datastructures/ThreadList.h"
+
+typedef struct {
+    ThreadList *threadList;
+    pthread_mutex_t mutex;
+} ThreadManager;
```

Comment:
Can we try to deduplicate at least `ThreadManager`, `ThreadList` and `Semaphore`? They're exactly the same in commix and immix

Reply:
I was thinking about this but now the way the plugin deals with the GCs would make it hard to share code as we compile all the code and then remove everything not selected.
https://github.com/scala-native/scala-native/blob/master/tools/src/main/scala/scala/scalanative/build/LLVM.scala#L127-L131

I think the way to improve the compile this is to only compile the code we are interested in. The only downside would be that we don't compile the code until the `GC` or `optional` code is selected and used. Not everything gets compiled. This also would require a PR and I would hate to slow progress here.

This is partly why I was suggesting that we totally leave `immix` alone since it is default so we can't mess up normal single threaded code and only enhance `commix` for now for multi-threaded. Since `commix is optional, we can mess people up if they are using it with single threaded but at least it would have less impact.

This would also simplify this PR as well with less changes.

Reply:
This is not correct what I say above. We only delete the not included `.o` files so that if we switch to another GC or remove a previous optional package the old code will be deleted. It still affects how we could share code between `immix` and `commix` however as it would be not really supported with the current code.